src/ape/managers/chain.py

```diff
@@ -408,18 +414,82 @@ def outgoing(self) -> Iterator[ReceiptAPI]:
         """
         All outgoing transactions, from earliest to latest.
         """
-        explorer = self.provider.network.explorer
-        explorer_receipts = (
-            [r for r in explorer.get_account_transactions(self.address)] if explorer else []
-        )
-        for receipt in explorer_receipts:
-            if receipt.sender != self.address:
-                # Likely ``self.address`` is a contract.
-                # Cache the receipts by their sender instead and skip them here.
-                self.chain_manager.history.append(receipt)
-                continue
+
+        start_nonce = 0
+        stop_nonce = self.__len__() - 1  # just to cache this value
+
+        # TODO: Add ephemeral network sessional history to `ape-cache` instead,
+        #       and remove this (replace with `yield from iter(self[:len(self)])`)
+        for receipt in self.sessional:
+            if receipt.nonce < start_nonce:
+                raise QueryEngineError("Sessional history corrupted")
+
+            if receipt.nonce > start_nonce:
+                # NOTE: There's a gap in our sessional history, so fetch from query engine
+                yield from iter(self[start_nonce : receipt.nonce + 1])  # noqa: E203
 
             yield receipt
+            start_nonce = receipt.nonce + 1  # start next loop on the next item
+
+        if start_nonce != stop_nonce:
+            # NOTE: there is no more sessional history, so just return query engine iterator
+            yield from iter(self[start_nonce : stop_nonce + 1])  # noqa: E203
+
+    def query(
```

Comment:
When I put at least 1 correct field mixed in with an incorrect one, it tells me a lot more information as to when I only put an incorrect one.

For example, when I do:

```
In [30]: account.history.query("transaction", "no", start_nonce=3, stop_nonce=6)
```

I get:

```
WARNING: Unrecognized field(s) 'no', must be one of {'return_value', 'gas_used', 'total_fees_paid', 'events', 'method_called', 'status', 'logs', 'txn_hash', 'ran_out_of_gas', 'transaction', 'block_number', 'trace', 'call_tree', 'contract_address', 'failed'}
Out[30]: 
                                         transaction
0  DynamicFeeTransaction:\n  chainId: 1\n  to: 0x...
1  DynamicFeeTransaction:\n  chainId: 1\n  to: 0x...
2  DynamicFeeTransaction:\n  chainId: 1\n  to: 0x...
3  DynamicFeeTransaction:\n  chainId: 1\n  to: 0x...
```

so it gives me the data and a warning, and that warning contains all the possible fields
(however, they could be displayed nicer, maybe without the curly brackets).

And then when I do this:

```
In [31]: account.history.query("no", start_nonce=3, stop_nonce=6)
```

I only get the error:

```
ValueError: No valid fields in ('no',).
```

Notice it does not tell me any of the correct fields.
I mainly bring this up because I thought it was interesting how the exception was better when I had at least 1 correct field, so I know we can do the same when there are 0 correct fields.

Reply:
Yeah, it would be nice to make these more consistent, but I think that can be a separate ticket

Reply:
https://github.com/ApeWorX/ape/pull/1346